pkg/ebpf/bytecode/runtime/protected_file.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package runtime
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/justincormack/go-memfd"
+)
+
+// This represent a symlink to a sealed ram-backed file
+type ProtectedFile interface {
+	Close() error
+	Reader() io.Reader
+	Name() string
+}
+
+type ramBackedFile struct {
+	symlink string
+	file    *memfd.Memfd
+}
+
+// This function returns a sealed ram backed file
+func NewProtectedFile(name, dir string, source io.Reader) (ProtectedFile, error) {
+	var err error
+
+	memfdFile, err := memfd.CreateNameFlags(name, memfd.AllowSealing|memfd.Cloexec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create memfd file: %w: ", err)
+	}
+	defer func() {
+		if err != nil {
+			memfdFile.Close()
+		}
+	}()
+
+	if source != nil {
+		if _, err = io.Copy(memfdFile, source); err != nil {
+			return nil, fmt.Errorf("error copying bytes to memfd file: %w", err)
+		}
+	}
+
+	// seal the memfd file, making it immutable.
+	if err = memfdFile.SetSeals(memfd.SealAll); err != nil {
+		return nil, fmt.Errorf("failed to seal memfd file: %w", err)
+	}
+
+	target := fmt.Sprintf("/proc/%d/fd/%d", os.Getpid(), memfdFile.Fd())
+	tmpFile := filepath.Join(dir, name)
+
+	link, _ := os.Readlink(target)
```

Comment:
Should we ignore this error?

Reply:
This function was called just for the debug log on the next line. I will both this and the log, since it was used just for validation during development.